### PR TITLE
Bfloat16 refactor

### PR DESCRIPTION
**Pull request by tsocha**
### PR types
Others

### PR changes
Others

### Describe
Refactor bfloat16 pass to keep it DRY.

### CODE REVIEW

File: paddle/fluid/framework/ir/mkldnn/cpu_bfloat16_placement_pass.cc [hunk 1]
```diff
@@ -50,58 +59,58 @@ void CPUBfloat16PlacementPass::SetMkldnnDataType(
...
+  int detected_operators = 0;
   auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
                      Graph* g) {
     GET_IR_NODE_FROM_SUBGRAPH(prev_out, prev_out, unsupported_bfloat16_pattern);
     GET_IR_NODE_FROM_SUBGRAPH(op, op, unsupported_bfloat16_pattern);
     if ((prev_out->Var()->GetDataType() != proto::VarType::FP32)) {
       op->Op()->SetAttr("mkldnn_data_type", std::string("float32"));
-      bfloat16_operators--;
+      PrettyLogDetail("---  demarked %s operator to bfloat16 ",
+                      op->Op()->Type());
```
Comment (tsocha) [inline paddle/fluid/framework/ir/mkldnn/cpu_bfloat16_placement_pass.cc]: These logs seems to be an debug leftover.
I plan to remove it.

File: paddle/fluid/framework/ir/mkldnn/cpu_bfloat16_placement_pass.cc [hunk 2]
```diff
@@ -50,58 +59,58 @@ void CPUBfloat16PlacementPass::SetMkldnnDataType(
...
   orphaned_bfloat16_pattern();
+  int detected_operators = 0;
   auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
                      Graph* g) {
     GET_IR_NODE_FROM_SUBGRAPH(op, op, orphaned_bfloat16_pattern);
 
     op->Op()->SetAttr("mkldnn_data_type", std::string("float32"));
-    bfloat16_operators--;
+    PrettyLogDetail("---  demarked %s operator to bfloat16 ", op->Op()->Type());
+    detected_operators++;
```
Comment (tsocha) [inline paddle/fluid/framework/ir/mkldnn/cpu_bfloat16_placement_pass.cc]: These logs seems to be an debug leftover.
I plan to remove it.

File: paddle/fluid/framework/ir/mkldnn/cpu_bfloat16_placement_pass.cc [hunk 3]
```diff
@@ -50,58 +59,58 @@ void CPUBfloat16PlacementPass::SetMkldnnDataType(
     if ((op->Op()->HasAttr("mkldnn_data_type") ||
          op->Op()->HasProtoAttr("mkldnn_data_type")) &&
         !platform::HasOpINT8DataType(op->Op())) {
+      PrettyLogDetail("---    marked %s operator to bfloat16 ",
+                      op->Op()->Type());
```
Comment (tsocha) [inline paddle/fluid/framework/ir/mkldnn/cpu_bfloat16_placement_pass.cc]: These logs seems to be an debug leftover.
I plan to remove it.

Comment (wozna) [reply paddle/fluid/framework/ir/mkldnn/cpu_bfloat16_placement_pass.cc]: It is helpful info while converting the model, so maybe you can change it to LOG(3) or LOG(4)

**Review by wozna — APPROVED** [review]
Very good job, LGTM

**Review by jczaja — APPROVED** [review]
LGTM

**Comment by tsocha** [comment]
@baoachun could you take a look at this?
Thank you.

**Review by baoachun — APPROVED** [review]
LGTM

**Pull request merged**